tests/rptest/tests/datalake/coordinator_retention_test.py

```diff
@@ -21,6 +21,7 @@
 
 class CoordinatorRetentionTest(RedpandaTest):
     def __init__(self, test_ctx, *args, **kwargs):
+        environment = {"__REDPANDA_TEST_DISABLE_BOUNDED_PROPERTY_CHECKS": "ON"}
```

Comment:
nitpick: I feel like we could push this down into `DatalakeServices` since it's the one starting redpanda. I guess the tedious part is already done tho 🤷

Reply:
Ah yea fair. Went ahead and did this so we don't have to keep thinking about this in future tests